File: api/src/main/java/com/cabinate/api/recipe/Recipe.java
package com.cabinate.api.recipe;

import java.time.Instant;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Document(collection = "recipes")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Recipe {

    @Id
    private String id;

    @Indexed
    private String title;
    private String description;
    private String sourceUrl;
    private String rawText;

    private Integer prepTimeMinutes;
    private Integer cookTimeMinutes;
    private Integer servings;

    private Instant createdAt;
    private Instant updatedAt;
}

File: mobile/src/types/recipe.ts
export interface Recipe {
  id: string;
  title: string;
  description?: string | null;
  sourceUrl?: string | null;
  rawText: string;
  prepTimeMinutes?: number | null;
  cookTimeMinutes?: number | null;
  servings?: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRecipeRequest {
  title: string;
  description?: string;
  sourceUrl?: string;
  rawText: string;
  prepTimeMinutes?: number;
  cookTimeMinutes?: number;
  servings?: number;
}

export interface UpdateRecipeRequest {
  title: string;
  description?: string;
  sourceUrl?: string;
  rawText: string;
  prepTimeMinutes?: number;
  cookTimeMinutes?: number;
  servings?: number;
}
